Framework/Algorithms/src/ModeratorTzero.cpp

```diff
@@ -218,8 +218,8 @@ void ModeratorTzero::exec() {
     }   // end of if(L2 >= 0)
 
     // Copy y and e data
-    outputWS->dataY(i) = inputWS->dataY(i);
-    outputWS->dataE(i) = inputWS->dataE(i);
+    outputWS->setSharedY(i, inputWS->sharedY(i));
+    outputWS->setSharedE(i, inputWS->sharedE(i));
```

Comment:
It would be nice to track the performance of algorithms that have significantly improved sharing. That would probably also make these often rather boring refactoring more  rewarding...? If you don't want to do it for all algorithms that are changed, maybe pick those that have a significant number of uses in the algorithm-usage-statistics?